allauth/socialaccount/providers/apple/apple_session.py

```diff
@@ -0,0 +1,35 @@
+from importlib import import_module
+
+from django.conf import settings
+from django.utils.cache import patch_vary_headers
+
+APPLE_SESSION_COOKIE_NAME = "apple-login-session"
+
+engine = import_module(settings.SESSION_ENGINE)
+SessionStore = engine.SessionStore
+
+def add_apple_session(request):
+    """
+    Fetch an apple login session
+    """
+    session_key = request.COOKIES.get(APPLE_SESSION_COOKIE_NAME)
+    request.apple_login_session = SessionStore(session_key)
+
+def persist_apple_session(request, response):
+    """
+    Save `request.apple_login_session` and set the cookie
+    """
+    patch_vary_headers(response, ('Cookie',))
+    request.apple_login_session.save()
+    response.set_cookie(
+        APPLE_SESSION_COOKIE_NAME,
+        request.apple_login_session.session_key,
+        max_age=None,
+        expires=None,
+        domain=settings.SESSION_COOKIE_DOMAIN,
+        # The cookie is only needed on this endpoint
+        path=response.url,
```

Comment:
I've done some digging and I think this is causing the issues with login on safari and presumably apple devices. `response.url` is a full URI whilst we should just be setting a path, I presume other browsers are more forgiving in their cookie handling.
 
I'll investigate further and submit a PR next week unless someone gets a chance sooner.

Reply:
Thanks, that would be amazing!

Reply:
The fix works for me – I left a note saying the same on the PR you later opened, thank you! :sparkling_heart:

Reply:
The fix works for me – I left a note saying the same on the PR you later opened, thank you! :sparkling_heart: hopefully others find it's good as well.